### PR TITLE
Create TranslateKernError for Darwin kernel error

### PR DESCRIPTION
**Pull request by kokarez**
I just add the likeliest errors, but we will need to add more in the future.

### CODE REVIEW

File: Sources/Host/Darwin/Platform.cpp [hunk 1]
```diff
@@ -118,6 +118,26 @@ std::string Platform::GetThreadName(ProcessId pid, ThreadId tid) {
...
+ErrorCode Platform::TranslateKernError(kern_return_t kret) {
+  switch (kret) {
+  case KERN_SUCCESS:
+    return ds2::kSuccess;
+  case KERN_FAILURE:
+    return ds2::kErrorUnknown;
+  case KERN_MEMORY_FAILURE:
+  case KERN_INVALID_ADDRESS:
+    return ds2::kErrorInvalidAddress;
+  case KERN_PROTECTION_FAILURE:
```
Comment (sas) [inline Sources/Host/Darwin/Platform.cpp]: `KERN_PROTECTION_FAILURE` should return `kErrorInvalidAddress`. `kErrorInvalidAddress` is used for everything that could be a segfault type failure, and `KERN_PROTECTION_FAILURE` seems to be one too.

See http://unix.superglobalmegacorp.com/xnu/newsrc/osfmk/mach/kern_return.h.html

File: Sources/Host/Darwin/Platform.cpp [hunk 2]
```diff
@@ -118,6 +118,26 @@ std::string Platform::GetThreadName(ProcessId pid, ThreadId tid) {
...
+  case KERN_NO_ACCESS:
+    return ds2::kErrorNoPermission;
+  case KERN_MEMORY_ERROR:
```
Comment (sas) [inline Sources/Host/Darwin/Platform.cpp]: Same for `KERN_MEMORY_ERROR`. `kErrorNoMemory` is only good for `KERN_RESOUCE_SHORTAGE` I think.

File: Sources/Host/Darwin/Platform.cpp [hunk 3]
```diff
@@ -118,6 +118,26 @@ std::string Platform::GetThreadName(ProcessId pid, ThreadId tid) {
...
+  case KERN_RESOURCE_SHORTAGE:
+    return ds2::kErrorNoMemory;
+  default:
+    DS2BUG("unknown error code: %s [%d]", mach_error_string(kret), kret);
```
Comment (sas) [inline Sources/Host/Darwin/Platform.cpp]: nit: can you do `"unknown error code: %d [%s]"` instead (and swap the two following arguments). In other similar `DS2BUG`s we only print the numeric value and I'll add the string version that the system returns to match the code you just wrote here.

File: Sources/Host/Darwin/Platform.cpp (hunk 1 above)
Comment (kokarez) [inline Sources/Host/Darwin/Platform.cpp]: Interesting, I was thinking kErrorInvalidAddress was really link a BAD_PARAM where the address is just wrong.

**Comment by sas** [comment]
I like this! 👍 

There are a couple of tiny changes that you should probably do before we check this in, I'll merge it after.

File: Sources/Host/Darwin/Platform.cpp (hunk 3 above)
Comment (kokarez) [inline Sources/Host/Darwin/Platform.cpp]: :+1:

**Comment by sas** [comment]
@facebook-github-bot shipit

**Pull request merged**